src/screens/Wallet.js

```diff
@@ -459,8 +461,13 @@ class Wallet extends React.Component {
 
     const renderUnlockedWallet = () => {
       let template;
-      if (tokenHistory.status === TOKEN_DOWNLOAD_STATUS.LOADING
-          || tokenBalance.status === TOKEN_DOWNLOAD_STATUS.LOADING) {
+      /**
+       * If an oldStatus of either balance or history should be undefined this
+       * means that this is the first time we are loading it, so it should show
+       * the loading message.
+       */
+      if (tokenHistory.oldStatus === undefined
```

Comment:
If the wallet loses connection with the full node and then reconnects. Will it reload the wallet and show the loading message? I think we should

Reply:
Changed to only show the entire wallet screen as loading if the wallet itself is in a loading state.